### ISSUE TITLE
Please honour the *_proxy environment variables

### ISSUE THREAD

**Issue by paddy-hack** [sitespeedio/browsertime#1494]
I guess I got spoiled by the many utilities that support `https_proxy`, `http_proxy` and `no_proxy` environment variables (or their uppercase cousins) out of the box.  I was surprised `sitespeed.io` doesn't and requires me to add command-line options to set the proxies.  These command-line option do not work when passed the typical values for the environment variables so these need some `sed` processing before they can be used as option values.  Moreover, support for `no_proxy` seems to be missing altogether, making it impossible to exempt certain locations from going through a proxy.

Searching the project here on `http_proxy` or `https_proxy` turns up several examples where people take the trouble to pass those environment variables to their Docker container, either via `docker`'s `-e` option or a Docker Compose file's `environment` attribute.  It would be so much easier for everyone if `sitespeed.io` honours these environment variables (and uses command-line arguments to override, preserving current behaviour).

I would prefer support for the lower case variant of the environment variables as those seem to be most widely supported in my experience (and [this blog post](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/)).  If you want to support the uppercase variants as well, I suggest uppercase → lowercase → command-line argument order in terms of increasing precedence.  

For `no_proxy`, please make sure to support CIDR notation because adding `10.0.0.0/8` one IP address at a time is no fun :wink:

**Comment by soulgalore** [comment]
Hi @paddy-hack I usually recommend people to test without a proxy to make sure it doesn't interfere with any metric, but if the implementation is simple and there are use cases for using a proxy,  I can review a PR :) Right now we use the default in WebDriver/Selenium. I moved the issue to Browsertime since it's cleaner if the implementation happens there.

**Comment by paddy-hack** [comment]
I can understand the desire to cut out factors you cannot control, like proxies, when doing development.  Yet, it is also important to know how one's *production* site is doing in terms of performance.

My issue arose when running tests against a(n example) production site in CI/CD, where the CI/CD runners/workers were behind (on the other side of) a proxy.  I have been dealing with proxies a fair deal and have gotten much used to things working out-of-the box if the `https_proxy`, `http_proxy` and `no_proxy` settings are correct.  For `sitespeed.io` that didn't work.

As for a PR, I'll leave that for someone with more free time on their hands.  In the mean time, I can work-around with a bit of `sed` magic and those "awkward" proxy-related command-line options you provide.  At least, until the need for `no_proxy` pops up its head ...

**Comment by paddy-hack** [comment]
> As for a PR, I'll leave that for someone with more free time on their hands.

Somehow, now I find myself looking at cooking something up.  More so after seeing #609 and #857 getting closed without being addressed.

**Comment by paddy-hack** [comment]
I've done a pile of additional testing and it appears that `sitespeed.io`/`browsertime` honours the various `*_proxy` environment variables (on Linux) after all.  I guess I got lost in all the moving parts of the CI/CD chain.

The Selenium webdriver uses the "system proxy settings" by default on Linux.  These appear to include `auto_proxy`, `https_proxy`, `http_proxy` and `no_proxy` as well as their uppercase counterparts.  I *think* `ftp_proxy` is supported too but haven't tested that.  Perhaps this is something that should be more prominently documented :thinking: 

I have tested using the `sitespeedio/sitespeed.io` Docker image (versions 14.1.0 and 16.9.2) passing environment variables via `docker run`'s `-e` option.  All results were as expected.  I should note that 16.9.2 gives better error feedback.

Sorry for the noise :bow:

As for that PR, I have something untested.  Guess I'll submit it just in case it's useful for non-Linux setups.
